### DISCUSSION TITLE
Run as Master?

### DISCUSSION THREAD

**Discussion by niknaknet** [Rodemfr/MicronetToNMEA#89] [General]
hi - great code!

I've been using micronetToNMEA to read from a tacktick at my sailing club, which works really well.

a challenge i've had has been that the head unit will go to sleep after a period of time - apparently it'll stay alive if it has a speed input, but as we're a stone ship, as it were, that's not really easy to do.

what i'm trying to do is spoof the head unit with one board, and read the traffic with the other.  I'm using the option 5 at the moment to read the raw local activity and an extension of the RF test to construct messages & adjust timings.  currently it's only trying to be simple and setup master (one of the teensy devices) with a masthead wind instrument, with ids etc hardcoded (as that's the limit of what i need to achieve)  

Although messages and timings match, i don't get any response from the masthead unit ... spoofing the device id of the head device & matching timings within single digits.  of course the mn100-2 connects with ease, and detects the ID collision!

do you have any details that might help in adding this function?

thanks

Nick

**Comment by Rodemfr** [comment]
Hi niknaknet,

First, I observe something similar than you : if a device does not see traffic on the micronet network, it will go to sleep mode after some time.
You probably already know it but all I know on the protocol is documented in this file : [micronet.txt](https://github.com/Rodemfr/MicronetToNMEA/blob/master/doc/Micronet.txt), just in case.

Concerning the master mode, I observed with my RTL-SDR dongle that when you start the system by powering up the master device, it first emits a long RF carrier for several seconds before sending commands. I did not check precisely the frequency of the carrier but it is probably the central frequency of Micronet's FSK modulation. This carrier is sent at full power because I see the amplitude is higher than standard communication.
I strongly suspect this carrier is here to activate a "Wake-up On Radio" functionality on the slave device. A special super low power mode which wake-up when activity on a given RF channel. This might be what you need to wake-up you mast unit.
